hub/core/storage/lru_cache.py

```diff
@@ -105,7 +114,25 @@ def get_hub_object(
         Returns:
             An instance of `expected_class` populated with the data.
         """
-
+        if get_partial:
+            assert issubclass(expected_class, BaseChunk)
```

Comment:
thought it is not a good practice to have asserts (which in case fail user would not understand what is going on)

Reply:
This is not a user-facing case, it would be caught during development.
"Asserts should be used to test conditions that should never happen" is the principle we've been using in the codebase.

Reply:
gotcha